### DISCUSSION TITLE
Does pnpm provide a module for parsing pnpm-lock.yaml?

### DISCUSSION THREAD

**Discussion by Plumbiu** [pnpm/pnpm#7034] [Q&A] [answered]
Recently I am developing a tool to display package dependencies. The method I use is to read the node_modules folder, but this is too time-consuming. Reading the pnpm-lock.yaml file is fast, but I don’t know how to use this file, so I would like to ask whether the pnpm community has provided such a method.

**Comment by weyert** [comment, marked as answer]
The specification explain the lock file format can be found here:
https://github.com/pnpm/spec/blob/master/lockfile/6.0.md

Also you might find https://github.com/pnpm/pnpm/tree/70b2830accacdef130112af258200ad705b57bae/lockfile/lockfile-file interesting

**Reply by gweesin** [reply]
Thanks, that's good for me!
